### PR TITLE
ruby: Add ruby-lsp as an experimental language server

### LINKED ISSUE #4834: shopify/ruby-lsp as an alternate language server
### Check for existing issues

- [X] Completed

### Describe the feature

I believe https://github.com/Shopify/ruby-lsp recently became [the recommended language server for VSCode](https://twitter.com/ShopifyEng/status/1641535539412082724?cxt=HHwWyICxjYf488ctAAAA). Wondering if there is a way to add it to Zed as an option. Ruby-lsp seems to be progressing quickly and would be a nice alternative to include.

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by vitallium**
Adds [ruby-lsp](https://shopify.github.io/ruby-lsp/) as an alternative LS for Ruby language.
While support for fully functional `ruby-lsp` is limited due to some limitations (see https://github.com/zed-industries/zed/pull/8613) I think it's OK to add it but disable by default. Thanks!

Resolves #4834.

Release Notes:

- N/A

### Some screenshots

Completion support
CleanShot 2024-05-13 at 22 58 23@2x

Symbol search
CleanShot 2024-05-13 at 23 03 59@2x

### CODE REVIEW

File: extensions/ruby/src/ruby.rs [hunk 1]
```diff
@@ -54,9 +69,29 @@ impl zed::Extension for RubyExtension {
             Solargraph::LANGUAGE_SERVER_ID => {
                 self.solargraph.as_ref()?.label_for_completion(completion)
             }
+            RubyLsp::LANGUAGE_SERVER_ID => self.ruby_lsp.as_ref()?.label_for_completion(completion),
             _ => None,
         }
     }
+
+    fn language_server_initialization_options(
```
Comment (vitallium) [inline extensions/ruby/src/ruby.rs:77]: This was taken from https://github.com/zed-industries/zed/pull/8613

**Review by maxdeviant — APPROVED** [review]
@vitallium Thanks!

**Comment by vitallium** [comment]
@maxdeviant, that was fast! You're welcome. And thank you for creating a separate PR to bump the version.

**Comment by andyw8** [comment]
That's great to hear, I've opened a PR to update the Ruby LSP docs:
https://github.com/Shopify/ruby-lsp/pull/2046

**Pull request merged**